mdoc-js/src/main/scala/mdoc/modifiers/JsModifier.scala

```diff
@@ -92,25 +81,30 @@ class JsModifier extends mdoc.PreModifier {
       case (Some(classpath), Some(scalacOptions)) =>
         config = JsConfig.fromVariables(ctx)
         reporter = ctx.reporter
-        val newClasspathHash = (classpath, scalacOptions, config.fullOpt).hashCode()
+        val compileClasspath = Classpath(classpath)
+        val linkerClasspath = Classpath(config.classpath)
+
+        val newClasspathHash =
+          (classpath, linkerClasspath, scalacOptions, config.fullOpt).hashCode()
         // Reuse the  linker and compiler when the classpath+scalacOptions haven't changed
         // to speed up unit tests by nearly 2x.
         if (classpathHash != newClasspathHash) {
-          linker = newLinker()
           classpathHash = newClasspathHash
           maybeCompiler = Some(new MarkdownCompiler(classpath, scalacOptions, target))
-          virtualIrFiles = {
-
-            val irContainer =
-              PathIRContainer.fromClasspath(Classpath(classpath).entries.map(_.toNIO))
-
-            val cache = irCache.newCache
+          val loader =
+            ScalaJSClassloader.create(linkerClasspath.entries.map(_.toURI.toURL()).toArray)
+          scalajsApi = Some(
+            loader
+              .loadClass("mdoc.js.worker.ScalaJSWorker")
```

Comment:
Can we use service loaders here instead of manual reflection? Service loaders have several tooling benefits, for example they are automatically understood by native-image (although that's not relevant here). https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html